lib/Yam/Agama/Pom/AgamaUpAndRunningPage.pm

```diff
@@ -29,7 +29,11 @@ sub new {
 
 sub expect_is_shown {
     my ($self) = @_;
-    assert_screen($self->{tag_array_ref_any_first_screen_shown}, $self->{timeout_expect_is_shown});
+    if (get_var('AGAMA_HEADLESS')) {
+        wait_serial('Connect to the Agama installer using these URLs:', 300) || die "Agama installer didn't start";
+    } else {
+        assert_screen($self->{tag_array_ref_any_first_screen_shown}, $self->{timeout_expect_is_shown});
+    }
```

Comment:
There is 3 problems here:
- Page Object do not have conditions, they are simple representation of what you see in the screen.
- This is a command, it is not really a screen that you need to map in any way.
- Avoid create more settings.

I think as in the future we might check for all architecture that agama is up and running via command line and remove the graphical way, we can just put this code in boot_agama.pm.
You can use a regex as condition for knowing if `EXTRABOOTPARAMS` contains `systemd.unit=multi-user.target`

Reply:
OK.

Reply:
Changed.